src/main/java/mike/Mike.java

```diff
@@ -0,0 +1,238 @@
+        }
+    }
+
+    /**
+     * Marks the task (specified by its index in the list) as "done".
+     *
+     * @param indexFromUser The index of the task (as seen by the user) to be marked in the list.
+     */
+    private void mark(int indexFromUser) {
+        ui.printReply(taskList.markInListWithMessage(indexFromUser));
+    }
+
+    /**
+     * Marks the task (specified by its index in the list) as "not done".
+     *
+     * @param indexFromUser The index of the task (as seen by the user) to be unmarked in the list.
+     */
+    private void unmark(int indexFromUser) {
+        ui.printReply(taskList.unmarkInListWithMessage(indexFromUser));
+    }
+
+    private void findInList(String searchWords) {
+        String listForPrint = taskList.findTasksInListForPrint(searchWords);
+        ui.printReply(listForPrint);
+    }
+
+    /**
+     * Processes the user's input then and responds accordingly.
+     *
+     * @param userInput String input from user as scanned in by Scanner.
+     */
+    private void processInput(String userInput) {
+        Parser parser = new Parser(userInput);
+
+        if (parser.isBlank()) {
+            noCharactersEnteredResponse();
+        }
+
+        String command = parser.getCommand();
+        try {
+            switch (command) {
+            case "list":
+                printList();
+                break;
+            case "mark":
+                int markIndex = parser.getIndex();
+                mark(markIndex);
+                break;
+            case "unmark":
+                int unmarkIndex = parser.getIndex();
+                unmark(unmarkIndex);
+                break;
+            case "remove":
+                int removeIndex = parser.getIndex();
+                removeTask(removeIndex);
+                break;
+            case "todo":
+                String todoName = parser.getTodoName();
+                addTodo(todoName);
+                break;
+            case "deadline":
+                String deadlineName = parser.getDeadlineName();
+                String deadlineDate = parser.getDeadlineDate();
+                addDeadline(deadlineName, deadlineDate);
+                break;
+            case "event":
+                String eventName = parser.getEventName();
+                String eventDate = parser.getEventDate();
+                addEvent(eventName, eventDate);
+                break;
+            case "find":
+                String searchWords = parser.getSearchWords();
+                findInList(searchWords);
+                break;
+            default:
+                String invalidCommandMessage =
+                        String.format("\n**Mike: I don't understand the command \"%s\"**",
+                                parser.getUserInput());
+                throw new UnsupportedOperationException(invalidCommandMessage);
+            }
+        } catch(UnsupportedOperationException e) {
```

Comment:
Maybe you can add whitespaces after your **catch** to be in accordance with the [Java Coding Standards](https://se-education.org/guides/conventions/java/intermediate.html).